api/src/main/java/org/pmiops/workbench/db/model/User.java

```diff
@@ -170,6 +171,18 @@ public void setAuthorities(Set<Authority> newAuthorities) {
     this.authorities = newAuthorities;
   }
 
+  @ElementCollection(fetch = FetchType.EAGER)
```

Comment:
I'm concerned about this - do we really always want to get the page visits for every user call? You have a specific dao call to get the user with visits. Also, please confirm that I'm not wrong here - my JPA is rusty.

Reply:
I initially tried to set this as FetchType.LAZY (if my understanding is correct - remember this is all new to me), but it kept failing. I can't remember the exact error, but when I dug around on it, seems like the fix was to change it to FetchType.EAGER. If someone knows how to do this differently (AND have it work), please speak up!

Reply:
I banged my head against this issue for a few days a while back. See https://github.com/all-of-us/workbench/pull/387

IIRC: simply doing a size() or Hibernate.initialize() will fetch a lazy collection on our dev server or when deployed to GAE. However - this won't work in tests as we currently have them configured b/c the EntityManager gets closed after the initial fetch. In a real spring server, we're using this setting (dubiously, it's the default) `spring.jpa.open-in-view=true` which leaves the EntityManager open for the duration of the request. So I think in theory another way of doing this would be to have a service method of `@Transactional` which just calls `getPageVisits()`.

But anyways, you already added an extra method to your UserDao, which is similar to what I did. So using EAGER here shouldn't be necessary. If you were seeing exceptions I suspect the issue may be in the other line I commented on (using the wrong `user` variable).

Reply:
Thank you! this was exactly it!